core/src/scene/sceneLayer.cpp

```diff
@@ -1,9 +1,12 @@
 #include "scene/sceneLayer.h"
 
 #include <algorithm>
+#include <type_traits>
 
 namespace Tangram {
 
+static_assert(std::is_move_constructible<SceneLayer>::value, "check");
+
 SceneLayer::SceneLayer(std::string _name, Filter _filter,
                        std::vector<DrawRuleData> _rules,
                        std::vector<SceneLayer> _sublayers,
```

Comment:
Should be a rvalue reference here for `_sublayers`?

Reply:
I believe the vector's contents are still moved at each step (i.e. the backing array is not copied) and that's what matters here. Stepping through the debugger seems to confirm this. If we were more paranoid we could make SceneLayer non-copyable, but I don't think this case is common enough to warrant that.